docs/library/bluetooth.rst

```diff
@@ -0,0 +1,306 @@
+                                      )
+
+    The three value handles (``hr``, ``tx``, ``rx``) can be used with
+    ``gatts_read``, ``gatts_write``, and ``gatts_notify``.
+
+    *Note:* Advertising must be stopped before registering services.
+
+.. method:: BLE.gatts_read(value_handle)
+
+    Reads the local value for this handle (which has either been written by
+    ``gatts_write`` or by a remote central).
+
+.. method:: BLE.gatts_write(value_handle, data)
+
+    Writes the local value for this handle, which can be read by a central.
+
+.. method:: BLE.gatts_notify(conn_handle, value_handle, [data])
+
+    Notifies a connected central that this value has changed and that it should
+    issue a read of the current value from this peripheral.
+
+    If ``data`` is specified, then the that value is sent to the central as part
+    of the notification, avoiding the need for a separate read request.
+
+
+Central Role (GATT Client)
+--------------------------
+
+.. method:: BLE.gap_connect(addr_type, addr, scan_duration_ms=2000)
+
+    Connect to a peripheral.
+
+    On success, the ``_IRQ_PERIPHERAL_CONNECT`` event will be raised.
+
+.. method:: BLE.gap_disconnect(conn_handle)
+
+    Disconnect the specified connection handle.
+
+    On success, the ``_IRQ_PERIPHERAL_DISCONNECT`` event will be raised.
+
+.. method:: BLE.gattc_discover_services(conn_handle)
+
+    Query a connected peripheral for its services.
+
+    For each service discovered, the ``_IRQ_GATTC_SERVICE_RESULT`` event will be
+    raised.
+
+.. method:: BLE.gattc_discover_characteristics(conn_handle, start_handle, end_handle)
+
+    Query a connected peripheral for characteristics in the specified range.
+
+    For each characteristic discovered, the ``_IRQ_GATTC_CHARACTERISTIC_RESULT``
+    event will be raised.
+
+.. method:: BLE.gattc_discover_descriptors(conn_handle, start_handle, end_handle)
+
+    Query a connected peripheral for descriptors in the specified range.
+
+    For each descriptor discovered, the ``_IRQ_GATTC_DESCRIPTOR_RESULT`` event
+    will be raised.
+
+.. method:: BLE.gattc_read(conn_handle, value_handle)
+
+    Issue a remote read to a connected peripheral for the specified
+    characteristic or descriptor handle.
+
+    On success, the ``_IRQ_GATTC_READ_RESULT`` event will be raised.
+
+.. method:: BLE.gattc_write(conn_handle, value_handle, data)
+
+    Issue a remote write to a connected peripheral for the specified
+    characteristic or descriptor handle.
+
+    On success, the ``_IRQ_GATTC_WRITE_STATUS`` event will be raised.
+
+
+class UUID
+----------
+
+
+Constructor
+-----------
+
+.. class:: UUID(value)
+
+    Creates a UUID instance with the specified value. The value can either be:
+
+    - A 16-bit integer. e.g. ``0x2908``.
+    - A 128-bit UUID string. e.g. ``'6E400001-B5A3-F393-E0A9-E50E24DCCA9E'``.
+
+
+Constants
+---------
+
+.. data:: bluetooth.FLAG_READ
+
+.. data:: bluetooth.FLAG_WRITE
+
+.. data:: bluetooth.FLAG_NOTIFY
```

Comment:
These can be formatted via:

    .. data:: bluetooth.FLAG_READ
              bluetooth.FLAG_WRITE
              bluetooth.FLAG_NOTIFY